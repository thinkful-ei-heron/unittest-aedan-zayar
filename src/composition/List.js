import React from 'react'
import Card from './Card'

class List extends React.Component {
    render() {
        let Cards = this.props.cards.map(card => { 
            return (<Card key={card.id} title={card.title} message={card.content} />)
        })
         return (
            <section className="List">
                <header className="List-header">
                    <h2>{this.props.header}</h2>
                </header>
                <div className="List-cards">
                    {Cards}
                </div>
            </section>
         )}
}

export default List;