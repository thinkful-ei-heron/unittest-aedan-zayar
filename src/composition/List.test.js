import renderer from 'react-test-renderer';
import React from 'react';
import ReactDOM from 'react-dom';
import List from './List.js';


describe('List component', () => {
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<List key="Test title" header="Test message" cards={[
        { id: 'a', title: 'First card', content: 'lorem ipsum' },
        { id: 'b', title: 'Second card', content: 'lorem ipsum' },
        { id: 'c', title: 'Third card', content: 'lorem ipsum' }]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<List key="testkey" header="Test Header" cards={[
      { id: 'a', title: 'First card', content: 'lorem ipsum' },
      { id: 'b', title: 'Second card', content: 'lorem ipsum' },
      { id: 'c', title: 'Third card', content: 'lorem ipsum' }
    ]
    } />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});
