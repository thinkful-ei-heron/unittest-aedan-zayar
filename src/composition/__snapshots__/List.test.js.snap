exports[`List component renders the UI as expected 1`] = `
<section
  className="List">
  <header
    className="List-header">
    <h2>
      Test message
    </h2>
  </header>
  <div
    className="List-cards">
    <div
      className="Card">
      <button
        className="button">
        delete
      </button>
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card">
      <button
        className="button">
        delete
      </button>
      <h3>
        Second card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card">
      <button
        className="button">
        delete
      </button>
      <h3>
        Third card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
  </div>
</section>
`;
